### DISCUSSION TITLE
Autocompletion feature

### DISCUSSION THREAD

**Discussion by xinyuluo** [osm-search/Nominatim#2657] [Q&A] [answered]
Hi, I'm curious why Nominatim lacks an autocompletion feature because I saw multiple closed issues mentioning about it. Is it an intentional technical design, or something pending to be implemented?

Thank you.

**Comment by mtmail** [comment, marked as answer]
The autocompletion feature request is as old as Nominatim itself, at least 10 years. Lots of people want it, it's very hard to implement and do right with the current database structure and code logic. Since a related project (Photon) does autocomplete and some of the same maintainers work on both we usually point to Photon as alternative. Photon is using data Nominatim created. Maybe in some years both projects can/will merge.

**Reply by xinyuluo** [reply]
I see. Thank you @mtmail for your answer!
